test/rng_testsuite/extra_methods_tests/engines_methods.pass.cpp

```diff
@@ -0,0 +1,263 @@
+// -*- C++ -*-
+//===-- engines_methods.hpp ----------------------             -------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Testing of different engines' methods
+
+#include "support/utils.h"
+#include <iostream>
+
+#if _ONEDPL_BACKEND_SYCL
+#    include <vector>
+#    include <CL/sycl.hpp>
+#    include <oneapi/dpl/random>
+
+#    define SEED 777
+#    define N_GEN 960
+#    define MINSTD_A 48271
+#    define MINSTD_C 0
+#    define MINSTD_M 2147483647
+#    define MINSTD_MIN 1
+#    define MINSTD_MAX 2147483646
+#    define RANLUX24_BASE_W 24
+#    define RANLUX24_BASE_S 10
+#    define RANLUX24_BASE_R 24
+#    define RANLUX24_BASE_MIN 0
+#    define RANLUX24_BASE_MAX 16777215
+#    define RANLUX24_P 223
+#    define RANLUX24_R 23
+
+template <class Engine, int N>
+std::int32_t
+check_params(Engine& engine)
+{
+    if constexpr (std::is_same<Engine, oneapi::dpl::minstd_rand>::value)
+    {
+        return ((oneapi::dpl::minstd_rand::multiplier != MINSTD_A) ||
+                (oneapi::dpl::minstd_rand::increment != MINSTD_C) || (oneapi::dpl::minstd_rand::modulus != MINSTD_M) ||
+                (engine.min() != MINSTD_MIN) || (engine.max() != MINSTD_MAX));
+    }
+    if constexpr (std::is_same<Engine, oneapi::dpl::minstd_rand_vec<N>>::value)
+    {
+        return ((oneapi::dpl::minstd_rand_vec<N>::multiplier != MINSTD_A) ||
+                (oneapi::dpl::minstd_rand_vec<N>::increment != MINSTD_C) ||
+                (oneapi::dpl::minstd_rand_vec<N>::modulus != MINSTD_M) || (engine.min() != MINSTD_MIN) ||
+                (engine.max() != MINSTD_MAX));
+    }
+    if constexpr (std::is_same<Engine, oneapi::dpl::ranlux24_base>::value)
+    {
+        return ((oneapi::dpl::ranlux24_base::word_size != RANLUX24_BASE_W) ||
+                (oneapi::dpl::ranlux24_base::short_lag != RANLUX24_BASE_S) ||
+                (oneapi::dpl::ranlux24_base::long_lag != RANLUX24_BASE_R) || (engine.min() != RANLUX24_BASE_MIN) ||
+                (engine.max() != RANLUX24_BASE_MAX));
+    }
+    if constexpr (std::is_same<Engine, oneapi::dpl::ranlux24_base_vec<N>>::value)
+    {
+        return ((oneapi::dpl::ranlux24_base_vec<N>::word_size != RANLUX24_BASE_W) ||
+                (oneapi::dpl::ranlux24_base_vec<N>::short_lag != RANLUX24_BASE_S) ||
+                (oneapi::dpl::ranlux24_base_vec<N>::long_lag != RANLUX24_BASE_R) ||
+                (engine.min() != RANLUX24_BASE_MIN) || (engine.max() != RANLUX24_BASE_MAX));
+    }
+    if constexpr (std::is_same<Engine, oneapi::dpl::ranlux24>::value)

```

Comment:
As far as I remember we didn't raise the C++ standard version to C++17 for RNG. It should work for C++14 and C++11 as well

Reply:
changed to function overloading